Sources/Sentry/SentryProfiler.mm

```diff
@@ -479,10 +440,16 @@ + (void)useProcessInfoWrapper:(SentryNSProcessInfoWrapper *)processInfoWrapper
     _gCurrentProcessInfoWrapper = processInfoWrapper;
 }
 
-+ (void)useTimerWrapper:(SentryNSTimerWrapper *)timerWrapper
++ (void)useMetricTimerWrapper:(SentryNSTimerWrapper *)timerWrapper
+{
+    std::lock_guard<std::mutex> l(_gProfilerLock);
+    _gMetricTimerWrapper = timerWrapper;
+}
+
++ (void)useTimeoutTimerWrapper:(SentryNSTimerWrapper *)timerWrapper
 {
     std::lock_guard<std::mutex> l(_gProfilerLock);
-    _gCurrentTimerWrapper = timerWrapper;
+    _gTimeoutTimerWrapper = timerWrapper;
```

Comment:
We now have two timer wrappers here, one for metrics collection as before, and a new one for profile timeout. We'll use two separate instances.